splatnet2statink.py

```diff
@@ -239,12 +240,66 @@ def main():
 
 	return m_value, is_s, is_t, is_r, filename
 
+def upload_recent(s_flag, t_flag, debug):
+	'''Check JSON once for new battles and upload them.'''
+
+	# initial checks
+	data = load_json(False)
+	try:
+		results = data["results"] # all we care about
+	except KeyError:
+		if YOUR_COOKIE == "":
+			reason = "blank"
+		elif data["code"] == "AUTHENTICATION_ERROR":
+			reason = "auth"
+		else:
+			reason = "other" # server error or player hasn't battled before
+		gen_new_cookie(reason)
+		data = load_json(False)
+		try:
+			results = data["results"] # try again with correct tokens; shouldn't get an error now...
+		except: # ...as long as there are actually battles to fetch (i.e. has played online)
+			print("Cannot access SplatNet 2 without having played at least one battle online.")
+			exit(1)
+
+	try:
+		url = "https://app.splatoon2.nintendo.net/api/records"
+		records = requests.get(url, headers=app_head, cookies=dict(iksm_session=YOUR_COOKIE))
+		global gender
+		gender = json.loads(records.text)["records"]["player"]["player_type"]["key"]
+	except:
+		pass
+
+	battles = [] # 50 recent battles on splatnet
+
+	# if r_flag, check if there are any battles in splatnet that aren't on stat.ink
+	print("Checking if there are previously-unuploaded battles...")
+	printed = False
+	url  = 'https://stat.ink/api/v2/user-battle?only=splatnet_number&count=50'
+	auth = {'Authorization': 'Bearer ' + API_KEY}
+	resp = requests.get(url, headers=auth)
+	statink_battles = json.loads(resp.text) # 50 recent battles on stat.ink
+
+	for i, result in reversed(list(enumerate(results))): # reversed chrono order
```

Comment:
You used this elsewhere and I copied it here to get the `i == 0` behavior for the most recent battle.

Reply:
`enumerate()` provides both a counter for numbers (`for i in int i < 100…`) and things (`for string in str_list…`) . think of the two types of for loops in java and bundling them into one. technically it returns an iterator that gives { (0, thing[0]), (1, thing[1]), … }

Reply:
`enumerate()` provides both a counter for numbers (`for int i < 100…`) and things (`for string in str_list…`) . think of the two types of for loops in java and bundling them into one. afaik technically it returns an iterator that gives `{(0, thing[0]), (1, thing[1]), …}`